### PR TITLE
Test shmem

### PR DESCRIPTION
**Pull request by WeberJulia**
**_Describe your changes here:_**

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

- [ ] The author added a BSD statement to `doc/` (or already has one)
- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)
- [ ] New source/header files are properly added to the Makefiles
- [ ] New Datatypes are added to t8indent_custom_datatypes.txt
- [ ] The reviewer executed the new code features at least once and checked the results manually
- [ ] The code is covered in an existing or new test case
- [ ] New tests use the Google Test framework
- [ ] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

### CODE REVIEW

**Comment by WeberJulia** [comment]
compare PR #393

**Review by jmark — CHANGES_REQUESTED** [review]

File: test/t8_data/t8_gtest_shmem.cxx [hunk 1]
```diff
@@ -190,29 +100,25 @@ t8_test_sc_shmem_alloc (sc_MPI_Comm comm)
   /* Seed random number generator. */
   srand (0);
 
+  /* Checking shared memory type */
   for (int shmem_type_int = SC_SHMEM_BASIC;
        shmem_type_int < SC_SHMEM_NUM_TYPES; ++shmem_type_int) {
     const sc_shmem_type_t shmem_type = (sc_shmem_type_t) shmem_type_int;
     int                 intrasize, intrarank;
-    t8_productionf ("Checking shared memory type %s.\n",
```
Comment (jmark) [inline test/t8_data/t8_gtest_shmem.cxx:197]: Please convert to `t8_debugf`.

Comment (jmark) [reply test/t8_data/t8_gtest_shmem.cxx:197]: If you have a different opinion about this you can answer to my comment and explain. :)

File: test/t8_data/t8_gtest_shmem.cxx [hunk 2]
```diff
@@ -190,29 +100,25 @@ t8_test_sc_shmem_alloc (sc_MPI_Comm comm)
...
 
     /* setup shared memory usage */
     t8_shmem_init (comm);
     t8_shmem_set_type (comm, shmem_type);
 
 #if T8_ENABLE_MPI
     const sc_shmem_type_t control_shmem_type = sc_shmem_get_type (comm);
-    SC_CHECK_ABORT (shmem_type == control_shmem_type,
-                    "Setting shmem type not succesful.\n");
+    EXPECT_EQ(shmem_type, control_shmem_type) << "Setting shmem type not succesfull.";
```
Comment (jmark) [inline test/t8_data/t8_gtest_shmem.cxx:115]: Should be an assertion.

File: test/t8_data/t8_gtest_shmem.cxx [hunk 3]
```diff
@@ -242,16 +146,13 @@ t8_test_sc_shmem_alloc (sc_MPI_Comm comm)
            0                    /* If neither SC_ENABLE_MPIWINSHARED or __bgq__ are defined, we neet a condition in this if
                                  * but do not want to execute any code. Hence, 0. */
         ) {
-        SC_CHECK_ABORT (intrarank == 0,
-                        "Type is one of window, window_prescan, bgq or bgq_prescan and this process should not have write permissions.\n");
+        EXPECT_EQ(intrarank, 0);
```
Comment (jmark) [inline test/t8_data/t8_gtest_shmem.cxx:149]: Make it an assertion and please add the message..

File: test/t8_data/t8_gtest_shmem.cxx [hunk 4]
```diff
@@ -242,16 +146,13 @@ t8_test_sc_shmem_alloc (sc_MPI_Comm comm)
...
       }
-      t8_productionf ("I have write permissions\n");
       shared_int[0] = random_number;
     }
     sc_shmem_write_end (shared_int, comm);
 
-    SC_CHECK_ABORTF (shared_int[0] == random_number,
-                     "shared integer was not assigned correctly at position 0.");
+    EXPECT_EQ(shared_int[0], random_number);
```
Comment (jmark) [inline test/t8_data/t8_gtest_shmem.cxx:155]: Make it an assertion and please add the message..

File: test/t8_data/t8_gtest_shmem.cxx [hunk 5]
```diff
@@ -242,16 +146,13 @@ t8_test_sc_shmem_alloc (sc_MPI_Comm comm)
            0                    /* If neither SC_ENABLE_MPIWINSHARED or __bgq__ are defined, we neet a condition in this if
                                  * but do not want to execute any code. Hence, 0. */
         ) {
-        SC_CHECK_ABORT (intrarank == 0,
-                        "Type is one of window, window_prescan, bgq or bgq_prescan and this process should not have write permissions.\n");
+        EXPECT_EQ(intrarank, 0);
       }
-      t8_productionf ("I have write permissions\n");
```
Comment (jmark) [inline test/t8_data/t8_gtest_shmem.cxx:248]: Conver to `t8_debug_f`.

**Review by jmark — CHANGES_REQUESTED** [review]
I highly recommend to alwasy use assertions when converting any `SC_CHECK_ABORT` calls. That way the test stops early when there are problems. Otherwise we risk segfaults or other nasty errors.

**Review by jmark — CHANGES_REQUESTED** [review]
There is still a change request left. Maybe you forgot it?

**Review by jmark — APPROVED** [review]

**Pull request merged**